### DISCUSSION TITLE
False positives sensor triggers. Interference

### DISCUSSION THREAD

**Discussion by zhvial** [micbay/RaceTimerAndController#6] [General]
I've had a problem with sensing laps. I have a simple microswitch with lever that is pushed by track original mechanism. Everything worked fine, until I've started to race. All my wires were routed underneath the track close to track lines. Every time I've pushed a trigger on the controller  timer started to count laps one time a second (debounce time is 1 sec.).
This happened because of interference between  wires that went to microswitch and race lines. Since we are sensing interrupts on analog lines of Arduino - a small change in signal is enough to trigger interrupts. 
The problem was solved by adding small capacitors 100nF between ground and each signal wire.

**Comment by micbay** [comment]
Thanks for sharing the details of an arrangement you found which lead to false triggers, and a working solution.

The Pin Change Interrupts are fired off when any change in the signal level is detected. This can be very sensitive and it will be important to try to mitigate radiated or conducted electromagnetic noise around the signal lines.
I don't know how much of a change is considered a trigger, but I suspect that, since the analog-to-digital convertors for the analog pins on the Nano have a 10bit resolution, the change threshold is around 5V/1024bits, or just under +/- 4.9mV.

Folks having issues with false triggers can try to use some of these typical noise reduction techniques:
- Use a bypass capacitor between the signal and ground (the solution described above). ([basic ref](http://www.learningaboutelectronics.com/Articles/What-is-a-bypass-capacitor.html)) ([advanced ref](https://www.renesas.com/us/en/document/apn/an1325-choosing-and-using-bypass-capacitors))
- Add a [low pass, high pass, or band pass filter ](https://www.arrow.com/en/research-and-events/articles/using-capacitors-to-filter-electrical-noise) using values of R and C that suppress frequencies of issue.
- Minimized the travel distance of the signal leads.
- Form long active and return leads into [twisted pairs](https://audiouniversityonline.com/twisted-pairs/).
- [Using ferrites](https://article.murata.com/en-us/article/basics-of-noise-countermeasures-lesson-8)
- Use [shielded leads](https://www.azosensors.com/article.aspx?ArticleID=724) (ideally shielding is grounded)
- Make sure everything is well grounded and isolated from radiated and conducted noise and [avoid ground loops](https://www.bapihvac.com/application_note/avoiding-ground-loops-application-note/).
